pandas/indexes/multi.py

```diff
@@ -374,9 +374,9 @@ def copy(self, names=None, dtype=None, levels=None, labels=None,
             labels = labels if labels is not None else deepcopy(self.labels)
             names = names if names is not None else deepcopy(self.names)
         else:
-            levels = self.levels
-            labels = self.labels
-            names = self.names
+            levels = levels if levels is not None else self.levels
```

Comment:
nit: it's an extra line, but less code to write:

    if levels is None:
        levels = self.levels